vyper/codegen/memory_allocator.py

```diff
@@ -110,6 +110,13 @@ def _expand_memory(self, size: int) -> int:
         before_value = self.next_mem
         self.next_mem += size
         self.size_of_mem = max(self.size_of_mem, self.next_mem)
+
+        if self.size_of_mem >= 2**64:
+            # this should not be caught
+            raise MemoryAllocationException(
+                "Tried to allocate {self.size_of_mem} bytes! (limit is 2**32 bytes)"
```

Comment:
probably should make it a constant and use it here as an f-string (and in the if statement)

Reply:
oh, right, i wanted to do it that way in the error message because `2**32` is easier to read than 4294967296.

Reply:
> oh, right, i wanted to do it that way in the error message because `2**32` is easier to read than 4294967296.

do the constant as a string and then parse it to int lol

Reply:
i think `eval()` is a bit overkill here :P unless you are thinking there is a cleaner way of parsing the string?

Reply:
or do log2

Reply:
i don't know about that. that results in an error message like
```
vyper.exceptions.MemoryAllocationException: Tried to allocate 18446744073709551712 bytes! (limit is 2**64.0 bytes)
```
and since floating point math is not very precise, we get the same error message for instance with `_ALLOCATION_LIMIT = 2**64 - 1`. so there is still the problem of drift between the error message and the value.

Reply:
```python
In [5]: MEM_LIMIT = 2**64

In [6]: f"2**{math.log2(MEM_LIMIT):2.0f}"
Out[6]: '2**64'
```